x/market/keeper/keeper.go

```diff
@@ -114,23 +110,49 @@ func (k *Keeper) createExecutionPlan(ctx sdk.Context, SourceDenom, DestinationDe
 	return bestPlan
 }
 
-func (k *Keeper) NewMarketOrderWithSlippage(ctx sdk.Context, srcDenom string, dst sdk.Coin, maxSlippage sdk.Dec, owner sdk.AccAddress, timeInForce types.TimeInForce, clientOrderId string) (*sdk.Result, error) {
+// GetSrcFromSlippage expresses the maximum source amount to spend to buy the
+// requested dst amount. Taking the corresponding source amount
+// (dst amount/last market price) and adding the slippage percentage is the
+// resulting value. The maxSlippage decimal expresses a percentage (1 is 100%).
+func (k *Keeper) GetSrcFromSlippage(
+	ctx sdk.Context, srcDenom string, dst sdk.Coin, maxSlippage sdk.Dec,
+) (sdk.Coin, error) {
+	if err := sdk.ValidateDenom(srcDenom); err != nil {
+		return sdk.Coin{}, sdkerrors.Wrapf(types.ErrInvalidInstrument,"%s", err)
+	}
+
+	if !dst.IsValid() {
```

Comment:
This check is already performed in the `ValidateBasic` method of the `Msg`. It's ok to assume that basic validation like that has taken place before the keeper logic.